packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,51 +110,84 @@ const withInfiniteScroll = (
         showSeeMore: false,
       };
 
-      this.onItemsFetch = props.onItemsFetch;
+      this.props.dataSource.onDataChange(this.updateData);
       this.observer = new IntersectionObserver(this.handleIntersection, {
         threshold: 1,
       });
     }
 
     componentDidMount() {
-      if (this.sentinel) {
-        this.observer.observe(this.sentinel);
+      this.loadMore().then(newState => {
+        this.setState(newState);
+      });
+    }
+
+    componentWillReceiveProps(newProps) {
```

Comment:
We should avoid using `componentWillReceiveProps` if we can. Looks like we could use `getDerivedStateFromProps` instead here? If we do use `getDerivedStateFromProps` we should make sure that the peer depedency on `React` is correct

Reply:
moved to componentDidUpdate